### PR TITLE
change interface to add support of psr/http-message:2.0

### PR DESCRIPTION
**Pull request by AleksandrCherepanov**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
         "ext-json": "*",
         "ext-libxml": "*",
         "psr/http-factory": "^1.0",
-        "psr/http-message": "^1.1"
+        "psr/http-message": "^2.0"
```
Comment (arokettu) [inline composer.json:41]: Tests seem to pass with 1.1 too, maybe relax to "^1.1 | ^2.0"?

Comment (AleksandrCherepanov) [reply composer.json]: Hi, sorry for the delay. Seems reasonable, added

**Comment by akrabat** [comment]
Thanks.

**Comment by Donov4n** [comment]
Hello,

These changes prevent proper `Response` method chaining without triggering errors from static analysis tools...

I can no longer do:

```php
return $response
    ->withStatus(StatusCode::STATUS_OK)
    ->withFile($myFile);
```

because with this PR the return type of withStatus is not `: static` but `Psr\Http\Message\ResponseInterface`, and since `withFile()` is not part of that interface (but is defined in the `Slim\Http\Response` decorator), we get `Undefined method 'withFile'.` error.

Wouldn’t returning `: static` instead of  (PSR-7) `: ResponseInterface` have been enough to remain compliant?

**Pull request merged**